### PR TITLE
NO-ISSUE: Disable nightly images job deploy

### PR DESCRIPTION
**Pull request by ricardozanini**
Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [ ] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

In this PR, we disable the image deploy to our repositories to avoid conflicts with Apache. Later, we can reenable to allow productization to test internally the mirrors.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>

- <b>Prod tests</b>  
  Please add comment: <b>Jenkins (re)run [prod|Prod|PROD]</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: .ci/jenkins/config/branch.yaml
Comment (radtriste) [inline .ci/jenkins/config/branch.yaml:1]: This one is not useful. It is defined in `kogito-pipelines`

File: .ci/jenkins/dsl/jobs.groovy [hunk 1]
```diff
@@ -172,6 +172,9 @@ void setupBuildImageJob(JobType jobType, boolean prodCI = false) {
         AUTHOR_CREDS_ID: "${GIT_AUTHOR_CREDENTIALS_ID}",
         AUTHOR_TOKEN_CREDS_ID: "${GIT_AUTHOR_TOKEN_CREDENTIALS_ID}",
     ])
+    if (Utils.hasBindingValue(this, 'DISABLE_DEPLOY')) {
+        jobParams.env.put('DEPLOY_IMAGE', !Utils.getBindingValue(this, 'DISABLE_DEPLOY').toBoolean())
```
Comment (radtriste) [inline .ci/jenkins/dsl/jobs.groovy]: `DEPLOY_IMAGE` is already a parameter to the job. Should not be defined as env var...
I just noticed that you can simply put https://github.com/kiegroup/kogito-images/blob/main/.ci/jenkins/Jenkinsfile.deploy#L285 to `false` (or dependent on `DISABLE_DEPLOY` value that needs to be passed as env in `setupDeployJob` method) ... That should be enough

Comment (ricardozanini) [reply .ci/jenkins/dsl/jobs.groovy]: @radtriste instead, I wanna keep the Jenkins env config so we can reenable easily when the time comes. There's a function now checking the parameter and when creating the job we check for the default value in the env config.

File: .ci/jenkins/config/branch.yaml
Comment (radtriste) [inline .ci/jenkins/config/branch.yaml:1]: Please remove that file. It is useless :)

Comment (ricardozanini) [reply .ci/jenkins/config/branch.yaml:1]: So how do I read from binding then?

File: .ci/jenkins/Jenkinsfile.deploy [hunk 2]
```diff
@@ -371,6 +371,10 @@ boolean isDeployLatestTag() {
     return params.DEPLOY_WITH_LATEST_TAG
 }
 
+boolean isDeployImage() {
+    return params.DEPLOY_IMAGE ? params.DEPLOY_IMAGE.toBoolean() : false
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.deploy:375]: you need to define this parameter into the `setupDeployJob` method in `jobs.groovy` file

File: .ci/jenkins/config/branch.yaml
Comment (radtriste) [reply .ci/jenkins/config/branch.yaml:1]: The whole pipelines config, for all repos, is there: https://github.com/kiegroup/kogito-pipelines/blob/main/.ci/jenkins/config/branch.yaml

Comment (ricardozanini) [reply .ci/jenkins/config/branch.yaml:1]: Ah ok, and there we have `disable.deploy = true`, so the binding catches from there? What if I want to change only in the local repo?

File: .ci/jenkins/Jenkinsfile.deploy (hunk 2 above)
Comment (ricardozanini) [reply .ci/jenkins/Jenkinsfile.deploy:375]: But this parameter is already set in BuildImage

Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.deploy:375]: setupBuildImageJob -> Jenkinsfile.build-image
setupDeployJob -> Jenkinsfile.deploy

Comment (ricardozanini) [reply .ci/jenkins/Jenkinsfile.deploy:375]: Doesn't deploy call the build image?

Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.deploy:375]: it does but here you are trying to access a parameter which is not defined for the deploy job ... you need to define it

Comment (ricardozanini) [reply .ci/jenkins/Jenkinsfile.deploy:375]: @radtriste done.

**Review by radtriste — APPROVED** [review]
lgtm

**Comment by radtriste** [comment]
@ricardozanini @jstastny-cz feel free to merge

**Pull request merged**